pkg/cmd/server/crypto/crypto.go

```diff
@@ -455,31 +449,55 @@ func NewKeyPair() (crypto.PublicKey, crypto.PrivateKey, error) {
 }
 
 // Can be used for CA or intermediate signing certs
-func newSigningCertificateTemplate(subject pkix.Name) (*x509.Certificate, error) {
+func newSigningCertificateTemplate(subject pkix.Name, expireDays int, currentTime func() time.Time) *x509.Certificate {
+
+	var caLifetimeInDays = defaultCACertificateLifetimeInDays
+	if expireDays > 0 {
+		caLifetimeInDays = expireDays
+	}
+
+	if caLifetimeInDays > defaultCACertificateLifetimeInDays {
+		warnAboutCertificateLifeTime(subject.CommonName, defaultCACertificateLifetimeInDays)
```

Comment:
All the warnings should happen in the actual command, in `Validate` method, imho.